crates/README.md

```diff
@@ -0,0 +1,147 @@
+
+# Roc Internals
+
+Roc has different rust crates for various binaries and libraries. These are described below to help give you an overview. If you see something here that is out of date, please correct it or discuss it in the [Zulip chat](https://roc.zulipchat.com/).
+
+Use `cargo doc` to generate docs; e.g. ```cargo doc --package roc_ast --open```.
+
+## `ast/` - `roc_ast`
+
+Code to represent the [Abstract Syntax Tree](https://en.wikipedia.org/wiki/Abstract_syntax_tree) as used by the editor.
+In contrast to the compiler, the types in this AST do not keep track of the location of the matching code in the source file.
+
+
+## `cli/` - `roc_cli`
+
+The `roc` binary that brings together all functionality in the Roc toolset.
+
+## `cli_utils/` - `cli_utils`
+
+Provides shared code for cli tests and benchmarks.
+
+## `code_markup/` - `roc_code_markup`
+
+Roc's very own [markup language](https://en.wikipedia.org/wiki/Markup_language). This library is used by the editor to display good-looking roc code. 
+
+## `compile/`
+
+Compiles `.roc` files and combine them with their platform into an executable binary. See [compiler/README.md](./compiler/README.md) for more information.
+
+TODO explain what "compiler frontend" is
+TODO explain what "compiler backend" is
+
+The compiler includes the following sub-crates;
+- `roc_alias_analysis` TODO - Need assistance
+- `arena-pool` TODO - Need assistance
+- `roc_build` TODO - Need assistance
+- `roc_builtins` provdes the Roc functions and modules that are implicitly imported into every module. See [README.md](./compiler/builtins/README.md) for more information.
+- `roc_can` TODO - Need assistance
+- `roc_collections` TODO - Need assistance
+- `roc_constrain` TODO - Need assistance
+- `roc_debug_flags` TODO - Need assistance
```

Comment:
Environment variables that can be toggled to aid debugging of the compiler itself.